docs/ref/settings.txt

```diff
@@ -632,23 +632,29 @@ default port. Not used with SQLite.
 
 Default: ``None``
 
-A string representing the time zone for datetimes stored in this database
-(assuming that it doesn't support time zones) or ``None``. This inner option of
-the :setting:`DATABASES` setting accepts the same values as the general
-:setting:`TIME_ZONE` setting.
+A string representing the time zone for this database connection or ``None``.
+This inner option of the :setting:`DATABASES` setting accepts the same values
+as the general :setting:`TIME_ZONE` setting.
+
+This option is only allowed when :setting:`USE_TZ` is ``True``. It is an error
+to set it when :setting:`USE_TZ` is ``False``.
+
+If the database backend doesn't support time zones (e.g. SQLite, MySQL,
+Oracle), Django reads and writes datetimes in local time according to this
+option if it is set and in UTC if it isn't.
 
 This allows interacting with third-party databases that store datetimes in
 local time rather than UTC. To avoid issues around DST changes, you shouldn't
 set this option for databases managed by Django.
 
-When :setting:`USE_TZ` is ``True`` and the database doesn't support time zones
-(e.g. SQLite, MySQL, Oracle), Django reads and writes datetimes in local time
-according to this option if it is set and in UTC if it isn't.
+If the database backend supports time zones (e.g. PostgreSQL), this option
+helps when using functions that depend on the local time zone to return the
+expected results.
```

Comment:
I'm not quite following this... 🤔

* The setting here says, "Use this TZ to read/write datetimes from this DB". 
* **Don't Use** if `USE_TZ = False`. 
* **Don't Use** if Django manages this DB.
* TZs not supported: read/write using this TZ, or UTC if not set. (OK, an unmanaged X DB, where everything is stored in `Europe/Madrid`, say. I'm `US/Chicago` and need it all mapped correctly.)

Is that right so far?

Then: 

* TZs supported: ??? An unmanaged Postgres, where everything is stored in `Europe/Madrid`, and ... ? 

I'm not sure it's clear what the behaviour is, or _when_ I **should** set this.

Reply:
Let's leave aside `USE_TZ = False` — this setting doesn't apply, all datetimes are naive, everything is easy (and you can't do anything about DST-related bugs or support multiple time zones).

In the rest of this comment, assume that `USE_TZ = True` and that all datetimes manipulated in Python are time zone aware datetimes.

> The setting here says, "Use this TZ to read/write datetimes from this DB".

In fact this covers two situations.

1. When the database backend doesn't support time zones _(I'm saying "database backend" because it could be the database itself or the DB-API connector or the Django database backend)_

    - when writing a datetime to the database, it's converted to the time zone of the database (controlled by the setting we're discussing here), the time zone information is stripped, and a naive datetime is sent to the database
    - when reading a datetime from the database, a naive datetime is obtained, then the time zone information is tacked on this datetime to make it aware (again, controlled by this setting)

    **In this situation, Django writes and reads different values in the database when the setting has different values.**

2. When the database backend supports time zones

    - when writing a datetime to the database, nothing special happens — the DB-API connector receives an aware datetime and deals with it
    - when reading a datetime from the database, nothing special needs to happen — the DB-API connector returns an aware datetime; however, for consistency, Django puts a pytz timezone instead the time zone information generated by psycopg2, which uses a fixed offset

    **In this situation, Django writes the same values in the database regardless of the value of the setting. It reads the same or different values, depending on how you look at it: it's always the same point in time but it's expressed in a different time zone.**

    Furthermore, if you're doing things like truncating datetimes to dates and aggregating them in a database query, you can do that in the correct time zone instead of UTC. This is a valid use case and an actual downside of `USE_TZ = True` vs. `USE_TZ = False` even for single time zone websites.

----

I didn't want to go into this level of details into the documentation of the setting, but perhaps I should?

Reply:
(I'll have a read/think, but I was thinking more towards, yes... 🙂)

Reply:
Or maybe an extended discussion in `databases.txt`

(Anyhow, let me have a think with the extended text here in mind.)

Reply:
> Furthermore, if you're doing things like truncating datetimes to dates and aggregating them in a database query, you can do that in the correct time zone instead of UTC. This is a valid use case and an actual downside of USE_TZ = True vs. USE_TZ = False even for single time zone websites.

So you're working in `Europe/Paris`. You want all `Event` objects on `SOME_DAY` but if you truncate in UTC you won't get the right results. Is that correct? 

Seems reasonable (but I'm immediately wondering about you wanting to work out the same, still from Paris, about Moscow...) 

(still thinking about it.)

Reply:
Yes, that's correct, if we're talking about truncating in the database with raw SQL.

Django APIs that truncate datetimes already do the right thing by converting datetimes to the correct time zone: https://github.com/django/django/blob/bc91f27a86090b4c688b56cd4e37f95eebe6e969/django/db/backends/postgresql/operations.py#L50-L53

If you want to do this in another time zone in raw SQL, then you need to do the same thing as Django in your SQL.

----

Really the most visible effect is that the ORM gives you datetimes in TIME_ZONE rather than in UTC. Django doesn't encourage this because it makes datetime arithmetic subtly unsafe by default — you need to normalize when you cross DST changes — while it's safe in UTC.

Reply:
Right, so I think we need to say that. 

I don't think the initial diff here reads very clearly, but a bit of a re-write should be able to phrase it better (as you have here) without needing to go into too much detail. (If that's not possible then I'd guess a small summary in the settings docs linking to a longer discussion elsewhere.) 

Thank you for taking the time with me. 🙂